substrate/primitives/runtime/src/generic/header.rs

```diff
@@ -77,7 +77,9 @@ where
 impl<Number, Hash> traits::Header for Header<Number, Hash>
 where
 	Number: BlockNumber,
+	<Number as codec::HasCompact>::Type: DecodeWithMemTracking,
```

Comment:
For this one I am note sure. If many people use the compact encoding of `BlockNumber` it would actually be handy to also directly bound it.

Right now `BlockNumber` implements `HasCompact` but its compact type doesn't implement `DecodeWithMemLimit` which make it not directly usable for call parameters.
I feel like `DecodeWithMemLimit` for a block number type is not too restrictive, and will be handy for users.

Reply:
Done